scripts/gui/CpStatus.lua

```diff
@@ -5,37 +5,56 @@ CpStatus = CpObject()
 function CpStatus:init(isActive, vehicle, currentWaypointIx, numberOfWaypoints)
     self:set(isActive, vehicle, currentWaypointIx, numberOfWaypoints)
     self.dirtyFlag = self.vehicle:getNextDirtyFlag()
+    self.remainingTime = CpRemainingTime(self.vehicle)
 end
 
-function CpStatus:set(isActive, vehicle, currentWaypointIx, numberOfWaypoints, timeRemaining)
+function CpStatus:set(isActive, vehicle, currentWaypointIx, numberOfWaypoints)
     self.isActive = isActive
     self.vehicle = vehicle
     self.currentWaypointIx = currentWaypointIx
     self.numberOfWaypoints = numberOfWaypoints
-    self.timeRemaining = timeRemaining
 end
 
 function CpStatus:reset()
     self.isActive = false
     self.currentWaypointIx = nil
     self.numberOfWaypoints = nil
-    self.timeRemaining = 0
+    self.remainingTime:reset()
+end
+
+function CpStatus:start()
+    self.remainingTime:start()
 end
 
 function CpStatus:setActive(active)
     if self.isActive ~= active then 
         self.isActive = active
         self:raiseDirtyFlag()
+        if not active then 
+            self:reset()
+        else 
+            self:start()
+        end
     end
 end
 
-function CpStatus:setWaypointData(currentWaypointIx,numberOfWaypoints,timeRemaining)
+function CpStatus:update(dt, isActive, strategy)
+    if isActive then 
+        if strategy then
+            strategy:updateCpStatus(self)
+            self.remainingTime:update(dt)
+        end
+    end 
+    self:setActive(isActive)
+end
+
+function CpStatus:setWaypointData(currentWaypointIx, numberOfWaypoints, course)
     if self.currentWaypointIx ~= currentWaypointIx then 
         self.currentWaypointIx = currentWaypointIx
         self.numberOfWaypoints = numberOfWaypoints
         self:raiseDirtyFlag()
+        self.remainingTime:calculate(course, currentWaypointIx)
```

Comment:
Why is the calculation (and the remainingTime object) in the HUD, why not in the fieldwork strategy, should the HUD really have a dependency on the course, and not just a stupid display of values provided by others?